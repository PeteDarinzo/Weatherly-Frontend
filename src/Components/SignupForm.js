import React, { useState } from 'react';
import { useHistory } from "react-router-dom";
import Box from '@mui/material/Box';
import TextField from '@mui/material/TextField';
import Button from '@mui/material/Button';



const SignupForm = ({ addUser }) => {

  const history = useHistory();

  const initialState = {
    username: "",
    password: "",
    zipCode: ""
  }

  const [formData, setFormData] = useState(initialState);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }));
  }

  function handleSubmit(e) {
    e.preventDefault();
    addUser({ ...formData });
    setFormData(initialState);
    history.push("/");
  }


  return (
    <div>
      <h1>Signup</h1>
      <Box
        component="form"
        onSubmit={handleSubmit}
        sx={{
          '& > :not(style)': { m: 1, width: '25ch' },
        }}
        noValidate
        autoComplete="off">
        <TextField name="username" id="username" label="Username" variant="standard" value={formData.username} onChange={handleChange} />
        <TextField name="password" id="password" label="Password" variant="standard" value={formData.password} onChange={handleChange} />
        <TextField name="zipCode" id="zip-code" label="Zip Code" variant="standard" value={formData.zipCode} onChange={handleChange} />
        <Button onClick={handleSubmit} variant="outlined">Submit</Button>
      </Box>
    </div>
  );
}

export default SignupForm;