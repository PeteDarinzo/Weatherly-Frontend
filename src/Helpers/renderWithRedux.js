import { render } from '@testing-library/react';
import { createStore, applyMiddleware } from "redux";
import { Provider } from 'react-redux';
import rootReducer from "../Reducers/rootReducer";
import thunk from "redux-thunk"


export default function renderWithRedux(component, { store = createStore(rootReducer, applyMiddleware(thunk)) } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        {component}
      </Provider>)
  }
}