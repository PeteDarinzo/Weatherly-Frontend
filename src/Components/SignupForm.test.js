import React from 'react';
import SignupForm from "./SignupForm";
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom"
import renderWithRedux from "../Helpers/renderWithRedux";

it("renders the signup form without crashing", () => {
  render(
    <MemoryRouter>
      <SignupForm register={() => { }} />
    </MemoryRouter>
  );
});

it("matches the signup form snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <SignupForm register={() => { }} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
