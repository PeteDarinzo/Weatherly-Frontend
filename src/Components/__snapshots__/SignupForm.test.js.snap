// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the signup form snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-pdaplq-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-6 css-rah4ii-MuiGrid-root"
      >
        <h3
          class="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
        >
          Sign Up
        </h3>
        <form>
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root makeStyles-field-2 css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="username"
                id="username-label"
              >
                Username
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="username"
                  name="username"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1g8cwq9"
                  >
                    <span>
                      Username *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root makeStyles-field-2 css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="password"
                id="password-label"
              >
                Password
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="password"
                  name="password"
                  required=""
                  type="password"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1g8cwq9"
                  >
                    <span>
                      Password *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root makeStyles-field-2 css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="postal-code"
                id="postal-code-label"
              >
                Postal Code
                <span
                  aria-hidden="true"
                  class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                >
                   *
                </span>
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="postal-code"
                  name="postalCode"
                  required=""
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1g8cwq9"
                  >
                    <span>
                      Postal Code *
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root makeStyles-field-2 css-wb57ya-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="country-code"
                id="country-code-label"
              >
                Country
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                variant="outlined"
              >
                <div
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby="country-code-label country-code"
                  class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                  id="country-code"
                  role="button"
                  tabindex="0"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name="countryCode"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1g8cwq9"
                  >
                    <span>
                      Country
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-jh47zj-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              Submit
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
